desc/objectives/objective_funs.py

```diff
@@ -484,11 +550,8 @@ def compute(self, *args, **kwargs):
 
     def compute_scalar(self, *args, **kwargs):
         """Compute the scalar form of the objective."""
-        if self.scalar:
-            f = self.compute(*args, **kwargs)
-        else:
-            f = jnp.sum(self.compute(*args, **kwargs) ** 2) / 2
-        return f
+        f = jnp.sum(self.compute(*args, **kwargs) ** 2) / 2
```

Comment:
Sorta Kinda. The issue is that when combining different objectives the blocked thing only works if everything is defined as sum of squares. 

For energy, the minima are the same, though the path taken by the optimizer might be different.

Reply:
Can you explain that first point a little?

Reply:
actually i might be wrong. I was thinking of an issue when combining scalar and vector objectives, but i think in that case we never call the compute_scalar method of the subobjective so it might be fine